src/gsy_e/models/area/scm_manager.py

```diff
@@ -196,81 +304,122 @@ def calculate_home_energy_bills(
-            else:
-                gsy_energy_bill = home_data.energy_need_kWh * market_maker_rate_decreased_fees
+            elif home_data.energy_need_kWh > FLOATING_POINT_TOLERANCE:
+                home_bill.set_bought_from_community(
+                    home_data.energy_need_kWh, market_maker_rate_decreased_fees,
+                    grid_fees * (1.0 - self._grid_fees_reduction), taxes_surcharges
+                )
                 home_data.create_buy_trade(
                     self._time_slot, DEFAULT_SCM_COMMUNITY_NAME, home_data.energy_need_kWh,
                     home_data.energy_need_kWh * market_maker_rate_decreased_fees
                 )
         else:
-            gsy_energy_bill = (
-                    home_data.allocated_community_energy_kWh * market_maker_rate_decreased_fees +
-                    (home_data.energy_need_kWh - home_data.allocated_community_energy_kWh) *
-                    market_maker_rate_normal_fees)
+            home_bill.set_bought_from_community(
+                home_data.allocated_community_energy_kWh, market_maker_rate_decreased_fees,
+                grid_fees * (1.0 - self._grid_fees_reduction), taxes_surcharges
+            )
+
+            energy_from_grid_kWh = (
+                    home_data.energy_need_kWh - home_data.allocated_community_energy_kWh)
+            home_bill.set_bought_from_grid(energy_from_grid_kWh, market_maker_rate_normal_fees,
+                                           grid_fees, taxes_surcharges)
+
             if home_data.allocated_community_energy_kWh > 0.0:
                 home_data.create_buy_trade(
                     self._time_slot, DEFAULT_SCM_COMMUNITY_NAME,
                     home_data.allocated_community_energy_kWh,
                     home_data.allocated_community_energy_kWh * market_maker_rate_decreased_fees
                 )
 
-            energy_from_grid_kWh = (
-                    home_data.energy_need_kWh - home_data.allocated_community_energy_kWh)
             if energy_from_grid_kWh > 0.:
                 home_data.create_buy_trade(
                     self._time_slot, DEFAULT_SCM_GRID_NAME,
                     energy_from_grid_kWh,
-                    energy_from_grid_kWh * market_maker_rate_decreased_fees
+                    energy_from_grid_kWh * market_maker_rate_normal_fees
                 )
 
-        savings_percent = (
-                ((base_energy_bill - gsy_energy_bill) / base_energy_bill) * 100.0
-                if base_energy_bill else 0.
-        )
-
-        self._bills[home_uuid] = HomeEnergyBills(
-            base_energy_bill=base_energy_bill,
-            gsy_energy_bill=gsy_energy_bill,
-            savings=(base_energy_bill - gsy_energy_bill),
-            savings_percent=savings_percent)
+        self._bills[home_uuid] = home_bill
+
+    def get_area_results(self, area_uuid: str, serializable: bool = False) -> Dict:
+        """Return the SCM results for one area (and one time slot)."""
```

Comment:
`and one time slot` means that these are the accumulated results until the current time slot. Right? 
(maybe worth to mention in the docstring IMO)

Reply:
No, these are results just from one time slot. Accumulation happens in gsy-framework (essentially before the simulation results are written to the database)

Reply:
Ah yes, the ScmManager is re-instantiated every market slot. Thanks! for the explanation!